src/main/php/PDepend/Source/AST/AbstractASTArtifact.php

```diff
@@ -295,18 +295,14 @@ public function getEndColumn(): int
         return $this->endColumn;
     }
 
-    /**
-     * @template T of array<string, mixed>|numeric-string
-     *
-     * @param T $data
-     * @return T
-     */
-    public function accept(ASTVisitor $visitor, $data = [])
+    public function accept(ASTVisitor $visitor): void
     {
         $methodName = 'visit' . substr(static::class, 22);
         $callable = [$visitor, $methodName];
-        assert(is_callable($callable));
-
-        return $callable($this, $data);
+        if (is_callable($callable)) {
+            $callable($this);
+        } else {
+            $visitor->visit($this);
+        }
```

Comment:
```suggestion
        if (is_callable($callable)) {
            $callable($this);

            return;
        }

        $visitor->visit($this);
```
Not using `else` is a PHPMD recommendation :)